packages/ketcher-core/src/application/editor/actions/rotate.ts

```diff
@@ -63,30 +63,59 @@ export function fromFlip(restruct, selection, dir, center) {
   })
 
   if (typeof isFragFound === 'number') {
-    return action // empty action
+    return flipPartOfStructure({
+      fids,
+      struct,
+      restruct,
+      dir,
+      action,
+      selection
+    })
   }
 
-  Object.keys(fids).forEach((frag) => {
-    const fragment = new Pile(fids[frag])
-
-    const bbox = struct.getCoordBoundingBox(fragment)
-    const calcCenter =
-      center ||
-      new Vec2((bbox.max.x + bbox.min.x) / 2, (bbox.max.y + bbox.min.y) / 2)
+  return flipStandaloneStructure({
+    fids,
+    struct,
+    restruct,
+    center,
+    dir,
+    action,
+    selection
+  })
+}
 
-    fragment.forEach((aid) => {
-      const atom = struct.atoms.get(aid)
-      const d = flipItemByCenter(atom, calcCenter, dir)
-      action.addOp(new AtomMove(aid, d))
-    })
+function getRotationPoint(struct, selection) {
+  const { bonds } = struct
+  const isAttachmentBond = ({ begin, end }) =>
+    (selection.atoms.includes(begin) && !selection.atoms.includes(end)) ||
+    (selection.atoms.includes(end) && !selection.atoms.includes(begin))
```

Comment:
Can we put these into separate variables?
e.g. `const atomsHaveBeginningBond = selection.atoms.includes(begin)`